apps/src/codegen.js

```diff
@@ -673,7 +673,7 @@ exports.selectCurrentCode = function (interpreter,
                                       editor,
                                       highlightClass) {
   var userCodeRow = -1;
-  if (interpreter.stateStack[0]) {
+  if (interpreter && interpreter.stateStack[0]) {
```

Comment:
I don't think this is related to my other changes, but I kept running into it while testing - when a parse error occurs, construction of the interpreter isn't complete yet so you get an exception here.